verifiable-db/src/query/public_inputs.rs

```diff
@@ -57,72 +96,118 @@ pub enum QueryPublicInputs {
     PlaceholderHash,
 }
 
+impl From<QueryPublicInputsUniversalCircuit> for QueryPublicInputs {
+    fn from(value: QueryPublicInputsUniversalCircuit) -> Self {
+        match value {
+            QueryPublicInputsUniversalCircuit::TreeHash => QueryPublicInputs::TreeHash,
+            QueryPublicInputsUniversalCircuit::OutputValues => QueryPublicInputs::OutputValues,
+            QueryPublicInputsUniversalCircuit::NumMatching => QueryPublicInputs::NumMatching,
+            QueryPublicInputsUniversalCircuit::OpIds => QueryPublicInputs::NumMatching,
+            QueryPublicInputsUniversalCircuit::LeftBoundaryRow => {
+                QueryPublicInputs::LeftBoundaryRow
+            }
+            QueryPublicInputsUniversalCircuit::RightBoundaryRow => {
+                QueryPublicInputs::RightBoundaryRow
+            }
+            QueryPublicInputsUniversalCircuit::MinPrimary => QueryPublicInputs::MinPrimary,
+            QueryPublicInputsUniversalCircuit::MaxPrimary => QueryPublicInputs::MaxPrimary,
+            QueryPublicInputsUniversalCircuit::SecondaryIndexValue => {
+                QueryPublicInputs::MinSecondary
+            }
+            QueryPublicInputsUniversalCircuit::PrimaryIndexValue => QueryPublicInputs::MaxSecondary,
+            QueryPublicInputsUniversalCircuit::Overflow => QueryPublicInputs::Overflow,
+            QueryPublicInputsUniversalCircuit::ComputationalHash => {
+                QueryPublicInputs::ComputationalHash
+            }
+            QueryPublicInputsUniversalCircuit::PlaceholderHash => {
+                QueryPublicInputs::PlaceholderHash
+            }
+        }
+    }
+}
+/// Public inputs for generic query circuits
+pub type PublicInputs<'a, T, const S: usize> = PublicInputsFactory<'a, T, S, false>;
+/// Public inputs for universal query circuit
+pub type PublicInputsUniversalCircuit<'a, T, const S: usize> = PublicInputsFactory<'a, T, S, true>;
```

Comment:
IIUC we should probably rename these like
* `PublicInputs` => `BatchingQueryPublicInputs` 
* `PublicInputsUniversalCircuit` => `TabularQueryPublicInputs` 

Would that be accurate ? The "generic query circuits" is not specific enough to be meaningful imo.

Reply:
I was not sure about using `BatchingQueryPublicInputs` because we no longer have the non-batched circuits, so imo doesn't make much sense to label these public inputs as `Batching` ones. Also, `TabularQueryPublicInputs` would be accurate now, but maybe once we will introduce generic tabular queries with results tree, I think we might still use `BatchingQueryPublicInputs`, so `TabularQueryPublicInputs` would be misleading. That's why I thought about using `UniversalCircuit` to identifiy this version of public inputs. Wdyt?

Reply:
For now I renamed `PublicInputs` in `PublicInputsQueryCircuits` in commit https://github.com/Lagrange-Labs/mapreduce-plonky2/pull/417/commits/0666bb9ba327edf45bd1899b89d63cf95036a179, until we find better names xD